rusk/src/lib/http/chain.rs

```diff
@@ -253,4 +263,47 @@ impl RuskNode {
 
         Ok(ResponseData::new(serde_json::to_value(stats)?))
     }
+
+    async fn simulate_tx(&self, tx: &str) -> anyhow::Result<ResponseData> {
+        let tx = hex::decode(tx).map_err(|e| {
+            anyhow::anyhow!("Failed to decode transaction: {e}")
+        })?;
+        let tx = ProtocolTransaction::from_slice(&tx)
+            .map_err(|e| anyhow::anyhow!("Invalid Data: {e:?}"))?;
+        let vm_handler = self.inner().vm_handler();
+        let vm_handler = vm_handler.read().await;
```

Comment:
Try to drop this as soon as possible, to not leave the read lock busy while executing the tx